src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,171 @@
+
+        // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document document = context.Document;
+            SyntaxNode root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+            SemanticModel model = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, context.CancellationToken).ConfigureAwait(false);
+            SyntaxGenerator generator = editor.Generator;
+            string title = MicrosoftNetCoreAnalyzersResources.AvoidConstArraysTitle;
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    title,
+                    async c => await ExtractConstArrayAsync(node, model, editor, generator, context.Diagnostics.First().Properties, c).ConfigureAwait(false),
+                    equivalenceKey: title),
+                context.Diagnostics);
+        }
+
+        private static async Task<Document> ExtractConstArrayAsync(SyntaxNode node, SemanticModel model, DocumentEditor editor,
+            SyntaxGenerator generator, ImmutableDictionary<string, string> properties, CancellationToken cancellationToken)
+        {
+            IArrayCreationOperation arrayArgument = GetArrayCreationOperation(node, model, cancellationToken, out bool isInvoked);
+            IEnumerable<ISymbol> typeFields = model.GetEnclosingSymbol(node.SpanStart, cancellationToken).ContainingType
+                .GetMembers().Where(x => x is IFieldSymbol);
+
+            // Get a valid member name for the extracted constant
+            string newMemberName = GetExtractedMemberName(typeFields.Select(x => x.Name), properties["paramName"]);
+
+            // Get method containing the symbol that is being diagnosed. Should always be in a method
+            IOperation? containingMethodBody = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody);
+            if (containingMethodBody is null)
+            {
+                // This is due to VB methods having a different structure than CS methods
+                containingMethodBody = arrayArgument.GetAncestor<IBlockOperation>(OperationKind.Block);
+            }
+
+            // Create the new member
+            SyntaxNode newMember = generator.FieldDeclaration(
+                newMemberName,
+                generator.TypeExpression(arrayArgument.Type),
+                GetAccessibility(model.GetEnclosingSymbol(containingMethodBody!.Syntax.SpanStart, cancellationToken)),
```

Comment:
The null suppression worries me a bit.

Reply:
I'll add a debug assertion to assert that it is not null, but my null check above should take care of all scenarios in the scope of the analyzer.